agent/include/content_analysis/sdk/analysis_agent.h

```diff
@@ -96,6 +96,17 @@ class ContentAnalysisEvent {
   // for debugging.
   virtual std::string DebugString() const = 0;
 
+  // Helper class to handle the lifetime and access of print data.
+  class ScopedPrintHandle {
+   public:
+    static std::unique_ptr<ScopedPrintHandle> Create(
```

Comment:
I think we should define all ctors and assignment operators to define this object as a move-only or copyable.  Both are possible so depends on what we want to impl (for example, it be copyable by duping the handle in the ctor).

Move-only seems to be all we need though.  If this is move-only, should this take a non-const reference to a `ContentAnalysisRequest` so that it can clear out the print data?  Thus preventing another scoped handle from being created?

Reply:
I think it makes sense to delete the copy constructor/operator, but I don't think it should own/take the request since it restricts a bit much how the class can be used, for instance it stops the print data from being read multiple times. I think we can just wait for feedback before making decisions about that class.